packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  InputFlexContainer,
+  useTheme,
+  SvgSearch,
+  SvgCloseSmall,
+  useSafeContext,
+  useId,
+  Icon,
+  useMergedRefs,
+} from '../utils';
+import type {
+  IconProps,
+  PolymorphicForwardRefComponent,
+  PolymorphicComponentProps,
+  InputFlexContainerProps,
+} from '../utils';
+import { IconButton } from '../Buttons/IconButton';
+import type { IconButtonProps } from '../Buttons/IconButton';
+import '@itwin/itwinui-css/css/input.css';
+
+const SearchBoxContext = React.createContext<
+  | {
+      /**
+       * Context prop for sizing subcomponents
+       */
+      size?: 'small' | 'large';
+      /**
+       * Context prop for disabling subcomponents
+       */
+      disabled?: boolean;
+      /**
+       * Id to pass to input
+       */
+      inputId: string;
+      setInputId: (inputId: string) => void;
+      /**
+       * Ref for input subcomponent
+       */
+      inputRef: React.RefObject<HTMLInputElement>;
+      /**
+       * Ref for open button subcomponent
+       */
+      openButtonRef: React.RefObject<HTMLButtonElement>;
+      /**
+       * Callback for closing expandable searchbox
+       */
+      onClose?: () => void;
+      /**
+       * Callback for expanding searchbox
+       */
+      onOpen?: () => void;
+    }
+  | undefined
+>(undefined);
+
+// ----------------------------------------------------------------------------
+
+type SearchBoxOwnProps = {
+  /**
+   * Whether the searchbox is expandable.
+   * @default false
+   */
+  expandable?: boolean;
+  /**
+   * Searchbox state toggle.
+   */
+  isExpanded?: boolean;
+  /**
+   * Function that is called on toggling (expand, collapse) searchbox.
+   */
+  onToggle?: (isExpanding: boolean) => void;
```

Comment:
this is probably a nit/subjective but i think instead of `isExpanding: boolean`, we could use a string union `state: 'expanded' | 'collapsed'`

Reply:
Do you have a hard preference on `onOpen` vs `onExpand`?

Reply:
lets do a poll in the team with a few options?

Reply:
Team voted Expand/collapse

Reply:
so change buttons to "ExpandButton" and "CollapseButton"?

Reply:
Changed to have `isExpanded`, `onExpand`, `onCollapse`, `SearchBox.ExpandButton` and `SearchBox.CollapseButton`